### ISSUE TITLE
STP agent - got stop request and removed the agent without any reason

### ISSUE THREAD

**Issue by yanpom** [thomsonreuters/CM-Well#570]
See following log:
During one of the STP config check, manager got no answer and from agent and stopped it immediatelly, then after getting an answer from sparql infoton restarted it - all statistics went lost as a result of it
We need to add a retry mechanism in such case (3 more times each 5 sec) because CMWELL can be busy sometimes and return no answer
```
2018-03-07 03:24:09,688 cm-well-ppe-akka.actor.default-dispatcher-2] INFO  BufferFillerActor [oafulloa-address] - no more data is available, will check again in 1 minute
2018-03-07 03:24:09,761 m-well-ppe-akka.actor.default-dispatcher-19] INFO  c.BufferFillerActor [oafulloa-phone] - no more data is available, will check again in 1 minute
2018-03-07 03:24:13,596 m-well-ppe-akka.actor.default-dispatcher-60] INFO  c.t.d.sparql.SparqlProcessorManager  - Checking the current status of the Sparql Triggered Processor manager config infotons
2018-03-07 03:24:17,128 m-well-ppe-akka.actor.default-dispatcher-19] INFO  c.t.d.sparql.SparqlProcessorManager  - Got stop and remove request for job [job: oafull, config name:None, sensors: oafulloa-names, oafulloa-identifiers, oafulloa-organization, oafulloa-address, oafulloa-relationship, oafulloa-hierarchy, oafulloa-industry, oafulloa-phone] from the user
2018-03-07 03:24:17,128 m-well-ppe-akka.actor.default-dispatcher-19] INFO  c.t.d.sparql.SparqlProcessorManager  - Stopping job [job: oafull, config name:None, sensors: oafulloa-names, oafulloa-identifiers, oafulloa-organization, oafulloa-address, oafulloa-relationship, oafulloa-hierarchy, oafulloa-industry, oafulloa-phone]. The job will actually stopped only after it will finish all its current operations
2018-03-07 03:24:17,128 [scala-execution-context-global-273]         INFO  c.t.d.sparql.SparqlProcessorManager  - job: [job: oafull, config name:None, sensors: oafulloa-names, oafulloa-identifiers, oafulloa-organization, oafulloa-address, oafulloa-relationship, oafulloa-hierarchy, oafulloa-industry, oafulloa-phone] finished successfully
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-23] INFO  c.t.d.sparql.SparqlProcessorManager  - Job [job: oafull, config name:None, sensors: oafulloa-names, oafulloa-identifiers, oafulloa-organization, oafulloa-address, oafulloa-relationship, oafulloa-hierarchy, oafulloa-industry, oafulloa-phone] has finished. Removing the job from the job list.
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-91] INFO  rFillerActor [oafulloa-organization] - BufferFillerActor I'm in post stop
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-86] INFO  c.BufferFillerActor [oafulloa-names] - BufferFillerActor I'm in post stop
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-40] INFO  erFillerActor [oafulloa-identifiers] - BufferFillerActor I'm in post stop
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-61] INFO  BufferFillerActor [oafulloa-address] - BufferFillerActor I'm in post stop
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-55] INFO  rFillerActor [oafulloa-relationship] - BufferFillerActor I'm in post stop
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-55] INFO  fferFillerActor [oafulloa-hierarchy] - BufferFillerActor I'm in post stop
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-61] INFO  ufferFillerActor [oafulloa-industry] - BufferFillerActor I'm in post stop
2018-03-07 03:24:17,129 m-well-ppe-akka.actor.default-dispatcher-61] INFO  c.BufferFillerActor [oafulloa-phone] - BufferFillerActor I'm in post stop
2018-03-07 03:24:43,597 m-well-ppe-akka.actor.default-dispatcher-86] INFO  c.t.d.sparql.SparqlProcessorManager  - Checking the current status of the Sparql Triggered Processor manager config infotons
2018-03-07 03:24:47,408 m-well-ppe-akka.actor.default-dispatcher-64] INFO  c.t.d.sparql.SparqlProcessorManager  - Got start request for job [job: oafull, config name:None, sensors: oafulloa-names, oafulloa-identifiers, oafulloa-organization, oafulloa-address, oafulloa-relationship, oafulloa-hierarchy, oafulloa-industry, oafulloa-phone] from the user
2018-03-07 03:24:56,461 m-well-ppe-akka.actor.default-dispatcher-64] INFO  c.t.d.sparql.SparqlProcessorManager  - starting job [job: oafull, config name:None, sensors: oafulloa-names, oafulloa-identifiers, oafulloa-organization, oafulloa-address, oafulloa-relationship, oafulloa-hierarchy, oafulloa-industry, oafulloa-phone]
2018-03-07 03:24:56,463 m-well-ppe-akka.actor.default-dispatcher-52] INFO  rFillerActor [oafulloa-organization] - starting BufferFillerActor
2018-03-07 03:24:56,463 m-well-ppe-akka.actor.default-dispatcher-71] INFO  c.BufferFillerActor [oafulloa-names] - starting BufferFillerActor
2018-03-07 03:24:56,463 [scala-execution-context-global-256]         INFO  c.Downloader [oafulloa-organization] - Got empty result. Waiting for 10 seconds before passing on the empty element.
2018-03-07 03:24:56,463 [scala-execution-context-global-252]         INFO  .t.d.d.c.Downloader [oafulloa-names] - Got empty result. Waiting for 10 seconds before passing on the empty element.
2018-03-07 03:24:56,464 m-well-ppe-akka.actor.default-dispatcher-52] INFO  BufferFillerActor [oafulloa-address] - starting BufferFillerActor
2018-03-07 03:24:56,464 [scala-execution-context-global-252]         INFO  .d.d.c.Downloader [oafulloa-address] - Got empty result. Waiting for 10 seconds before passing on the empty element.
2018-03-07 03:24:56,465 m-well-ppe-akka.actor.default-dispatcher-52] INFO  rFillerActor [oafulloa-relationship] - starting BufferFillerActor
2018-03-07 03:24:56,465 [scala-execution-context-global-252]         INFO  c.Downloader [oafulloa-relationship] - Got empty result. Waiting for 10 seconds before passing on the empty element.
2018-03-07 03:24:56,465 m-well-ppe-akka.actor.default-dispatcher-80] INFO  erFillerActor [oafulloa-identifiers] - starting BufferFillerActor
2018-03-07 03:24:56,465 m-well-ppe-akka.actor.default-dispatcher-91] INFO  fferFillerActor [oafulloa-hierarchy] - starting BufferFillerActor
2018-03-07 03:24:56,465 [scala-execution-context-global-256]         INFO  .d.c.Downloader [oafulloa-hierarchy] - Got empty result. Waiting for 10 seconds before passing on the empty element.
2018-03-07 03:24:56,465 [scala-execution-context-global-256]         INFO  .c.Downloader [oafulloa-identifiers] - Got empty result. Waiting for 10 seconds before passing on the empty element.
2018-03-07 03:24:56,466 m-well-ppe-akka.actor.default-dispatcher-60] INFO  ufferFillerActor [oafulloa-industry] - starting BufferFillerActor
2018-03-07 03:24:56,466 [scala-execution-context-global-256]         INFO  d.d.c.Downloader [oafulloa-industry] - Got empty result. Waiting for 10 seconds before passing on the empty element.
2018-03-07 03:24:56,467 cm-well-ppe-akka.actor.default-dispatcher-2] INFO  c.BufferFillerActor [oafulloa-phone] - starting BufferFillerActor
```

**Comment by nicklester** [comment]
Can we confirm this is still an issue. Currently if config infoton cannot be read:

1) Job will not run
2) However it will run on the next times iteration
3) Stats will not now be lost as they are persisted

**Comment by yanpom** [comment]
Still happens

**Comment by nicklester** [comment]
@yanpom The statistics should now be persisted due to PR #711. 
The retry of the config infotons call is covered in issue  #516.
Closing issue

**Comment by yanpom** [comment]
Didn't see it happen for a while, Closing meanwhile, and will reopen if the issue recreates
